Controller/Gateway/MoneroPayment.php

```diff
@@ -330,63 +336,129 @@ public function retriveprice($currency)
                 return $price;
         }
     }
-    
-    public function paymentid_cookie()
+
+    public function subaddress_cookie()
     {
-        if (!isset($_COOKIE['payment_id']))
-        {
-            $payment_id = bin2hex(openssl_random_pseudo_bytes(8));
-            setcookie('payment_id', $payment_id, time() + 2700);
+        if (!isset($_COOKIE['xmr_subaddress'])) {
+            // Generate subaddress
+            $xmr_subaddress = $this->monero_daemon->create_address(0);
+            setcookie('xmr_subaddress', $xmr_subaddress['address'], time() + 2700);
+        } else {
+            $xmr_subaddress = ['address' => $_COOKIE['xmr_subaddress']];
```

Comment:
Parse the address before set it to $xmr_subaddress otherwise it's an input that might be controlled by an attacker (nothing too sensitive, but..)

Reply:
sanitized and validated.. pls check